client/ayon_core/pipeline/usdlib.py

```diff
@@ -108,256 +256,426 @@ def create_shot(filepath, layers, create_layers=False):
-        append = Usd.ListPositionBackOfAppendList
-        variant_set = root_prim.GetVariantSets().AddVariantSet(
-            variantset, append
-        )
+    Args:
+        layer (Sdf.Layer): Layer to operate in.
+        prim_path (Sdf.Path): Prim path to reference into.
+            This may include variant selections to reference into a prim
+            inside the variant selection.
+        reference (Sdf.Reference): Reference to add.
+        order  (int): Order.
 
-        for variant, variant_path in variants:
+    Returns:
+        Sdf.PrimSpec: The prim spec for the prim path.
 
-            if default_variant is None:
-                default_variant = variant
+    """
+    assert isinstance(order, int), "order must be integer"
+
+    # Sdf.Reference is immutable, see: `pxr/usd/sdf/wrapReference.cpp`
+    # A Sdf.Reference can't be edited in Python so we create a new entry
+    # matching the original with the extra data entry added.
+    custom_data = reference.customData
+    custom_data["ayon_order"] = order
+    reference = Sdf.Reference(
+        assetPath=reference.assetPath,
+        primPath=reference.primPath,
+        layerOffset=reference.layerOffset,
+        customData=custom_data
+    )
 
-            variant_set.AddVariant(variant, append)
-            variant_set.SetVariantSelection(variant)
-            with variant_set.GetVariantEditContext():
-                _reference(variant_path)
+    # TODO: inherit type from outside of variants if it has it
+    prim_spec = get_or_define_prim_spec(layer, prim_path, "Xform")
 
-                log.info("Variants..")
-                log.info("Variant: %s" % variant)
-                log.info("Path: %s" % variant_path)
+    # Insert new entry at correct order
+    entries = list(prim_spec.referenceList.prependedItems)
 
-        if set_default_variant:
-            variant_set.SetVariantSelection(default_variant)
+    if not entries:
+        prim_spec.referenceList.prependedItems.append(reference)
+        return prim_spec
 
-    return stage
+    for index, existing_ref in enumerate(entries):
+        existing_order = existing_ref.customData.get("order")
+        if existing_order is not None and existing_order < order:
+            log.debug(
+                f"Inserting new reference at {index}: {reference}"
+            )
+            entries.insert(index, reference)
+            break
+    else:
+        prim_spec.referenceList.prependedItems.append(reference)
+        return prim_spec
 
+    prim_spec.referenceList.prependedItems[:] = entries
+    return prim_spec
 
-def get_usd_master_path(folder_entity, product_name, representation):
-    """Get the filepath for a .usd file of a product.
 
-    This will return the path to an unversioned master file generated by
-    `usd_master_file.py`.
+def set_variant_reference(sdf_layer, prim_path, variant_selections, path,
+                          as_payload=False,
+                          append=True):
```

Comment:
```suggestion
def set_variant_reference(
    sdf_layer,
    prim_path,
    variant_selections, path,
    as_payload=False,
    append=True
):
```

Reply:
Is there are reason you still kept `path` on the same line? (accidental?)

Reply:
accident